### PR TITLE
[client,X11] Fix RAILS modal dialog handling

### PR DESCRIPTION
**Pull request by akallabeth**
* Fixes Issue #10530 :
  * Update window style also if WINDOW_ORDER_STATE_NEW is not set
  * On window styleEx WS_EX_CONTROLPARENT | WS_EX_TOOLWINDOW | WS_EX_DLGMODALFRAME set as popup window of the parent window (XSetTransientForHint and XRestackWindows to ensure ordering)
  * On window styleEx WS_EX_TOPMOST | WS_EX_TOOLWINDOW set it modal (_NET_WM_STATE_ABOVE and XRaiseWindow)
* Add logging helper functions window_styles_to_string, window_styles_ex_to_string and x11_error_to_string
* Add new _NET_WM_STATE X11 Atom values (all defined by https://specifications.freedesktop.org/wm-spec/1.3/ar01s05.html )

### CODE REVIEW

File: client/X11/xfreerdp.h [hunk 1]
```diff
@@ -230,11 +230,18 @@ struct xf_context
 	Atom _NET_SUPPORTING_WM_CHECK;
 
 	Atom _NET_WM_STATE;
-	Atom _NET_WM_STATE_FULLSCREEN;
-	Atom _NET_WM_STATE_MAXIMIZED_HORZ;
+	Atom _NET_WM_STATE_MODAL;
+	Atom _NET_WM_STATE_STICKY;
 	Atom _NET_WM_STATE_MAXIMIZED_VERT;
+	Atom _NET_WM_STATE_MAXIMIZED_HORZ;
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply client/X11/xfreerdp.h:236]: addressed in follow up pr

File: client/X11/xf_window.c [hunk 2]
```diff
@@ -108,6 +112,127 @@ typedef struct
...
+		ENTRY(WS_VISIBLE);
+		ENTRY(WS_VSCROLL);
+		default:
+			return NULL;
+	}
+}
+
+const char* window_styles_to_string(UINT32 style, char* buffer, size_t length)
+{
+	_snprintf(buffer, length, "style[0x%08" PRIx32 "] {", style);
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply client/X11/xf_window.c:161]: addressed in follow up pr

**Comment by akallabeth** [comment]
@freerdp-bot test

File: client/X11/xfreerdp.h [hunk 3]
```diff
@@ -230,11 +230,18 @@ struct xf_context
 	Atom _NET_SUPPORTING_WM_CHECK;
 
 	Atom _NET_WM_STATE;
-	Atom _NET_WM_STATE_FULLSCREEN;
-	Atom _NET_WM_STATE_MAXIMIZED_HORZ;
+	Atom _NET_WM_STATE_MODAL;
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply client/X11/xfreerdp.h:233]: changed in follow up pr

**Review by nfedera — APPROVED** [review]

**Pull request merged**